libs/account-lib/src/services/provider-webhook.service.ts

```diff
@@ -40,6 +40,60 @@ export class ProviderWebhookService implements OnModuleDestroy {
       timeout: httpConfig.httpResponseTimeoutMS,
     });
 
+    // Add request interceptor for logging
+    this.webhook.interceptors.request.use(
+      (config) => {
+        this.logger.debug(`[Provider] Making ${config.method?.toUpperCase()} request to: ${config.baseURL}${config.url}`);
+        this.logger.debug('[Provider] Request config:', {
+          method: config.method,
+          url: config.url,
+          timeout: config.timeout,
+          headers: config.headers,
+          data: config.data
+        });
+        return config;
+      },
+      (error) => {
+        this.logger.error('[Provider] Request interceptor error:', error);
+        return Promise.reject(error);
+      }
+    );
+
+    // Add response interceptor for logging
+    this.webhook.interceptors.response.use(
+      (response) => {
+        this.logger.debug(`[Provider] Response received for ${response.config.method?.toUpperCase()} ${response.config.url}`);
+        this.logger.debug('[Provider] Response details:', {
+          status: response.status,
+          statusText: response.statusText,
+          headers: response.headers,
+          data: response.data
+        });
+        return response;
+      },
+      (error) => {
+        if (axios.isAxiosError(error)) {
+          this.logger.error('[Provider] Response interceptor error:');
```

Comment:
Seeing all this repeated, it feels like we should have a logger helper function.